runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -259,32 +290,42 @@ def GetCommand(): list<string>
 enddef
 
 def StartDebug(bang: bool, ...gdb_args: list<string>)
+  if g:termdebug_is_running == true
+    Echoerr('Terminal debugger already running, cannot run two')
+    return
+  endif
+
   InitScriptVariables()
+  if !SanityCheck()
+    return
+  endif
```

Comment:
Those checks here:
```
  if g:termdebug_is_running == true
    Echoerr('Terminal debugger already running, cannot run two')
    return
  endif

  InitScriptVariables()
  if !SanityCheck()
    return
  endif
```
are done in `StartDebug()` and `StartDebugCommand()`. Better move it directly into `StartDebug_internal()` so those 2 functions don't duplicate logic and stay those thin wrappers around `StartDebug_internal()`

Reply:
Usch! You are absolutely right!